### PR TITLE
Create KubernetesPatchJobOperator operator

### PR DESCRIPTION
**Pull request by MaksYermak**
In this PR I have created an operator `KubernetesPatchJobOperator` for updating Job.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/hooks/kubernetes.py [hunk 1]
```diff
@@ -527,6 +527,20 @@ def list_jobs_from_namespace(self, namespace: str) -> V1JobList:
         """
         return self.batch_v1_client.list_namespaced_job(namespace=namespace, pretty=True)
 
+    def patch_namespaced_job(self, job_name: str, namespace: str, body: object) -> V1Job:
+        """
+        Update the specified Job.
+
+        :param name: name of the Job
```
Comment (pankajastro) [inline airflow/providers/cncf/kubernetes/hooks/kubernetes.py]:
```suggestion
        :param job_name: name of the Job
```

File: airflow/providers/cncf/kubernetes/operators/job.py [hunk 2]
```diff
@@ -284,3 +286,71 @@ def reconcile_job_specs(
...
+    @cached_property
+    def hook(self) -> KubernetesHook:
+        return KubernetesHook(
+            conn_id=self.kubernetes_conn_id,
+            in_cluster=self.in_cluster,
+            config_file=self.config_file,
+            cluster_context=self.cluster_context,
+        )
+
+    def execute(self, context: Context):
```
Comment (pankajastro) [inline airflow/providers/cncf/kubernetes/operators/job.py]: shall we add return type here too to keep it consistent?

File: airflow/providers/cncf/kubernetes/hooks/kubernetes.py (hunk 1 above)
Comment (MaksYermak) [reply airflow/providers/cncf/kubernetes/hooks/kubernetes.py]: @pankajastro I fixed it, thanks!

File: airflow/providers/cncf/kubernetes/operators/job.py (hunk 2 above)
Comment (MaksYermak) [reply airflow/providers/cncf/kubernetes/operators/job.py]: @pankajastro good idea, I have added it.

**Review by pankajastro — APPROVED** [review]
LGTM

**Comment by VladaZakharova** [comment]
@potiuk 
Hi! Can we please merge this? Thanks!

**Pull request merged**